src/read.c

```diff
@@ -3418,11 +3466,11 @@ static avifBool avifTilesCanBeDecodedWithSameCodecInstance(avifDecoderData * dat
     }
     const uint8_t firstTileOperatingPoint = data->tiles.tile[0].operatingPoint;
     const avifBool firstTileAllLayers = data->tiles.tile[0].input->allLayers;
-    const uint8_t * firstTileType = data->tiles.tile[0].type;
+    const avifCodecType firstTileCodecType = data->tiles.tile[0].codecType;
     for (unsigned int i = 1; i < data->tiles.count; ++i) {
         const avifTile * tile = &data->tiles.tile[i];
-        if (tile->operatingPoint != firstTileOperatingPoint || tile->input->allLayers != firstTileAllLayers ||
-            memcmp(tile->type, firstTileType, 4)) {
+        if ((tile->operatingPoint != firstTileOperatingPoint) || (tile->input->allLayers != firstTileAllLayers) ||
+            (tile->codecType != firstTileCodecType)) {
```

Comment:
We do not have to support this feature, but I thought it would not hurt to do so.

I believe AV1 and HEVC tiles can be mixed in a HEIF grid if they share the same properties (subsampling etc.) but I doubt any decoder handles and tests that as of today. It is even less likely that an encoder allows that.

Reply:
I wasn't expecting any action on this comment, it was more of a general observation. But thanks for adding the check :)